bundle/edu.gemini.p1backend/TODO.md

```diff
@@ -0,0 +1,50 @@
+# Short term
+
+Something that improves the situation now. Nothing for the future. A dead-end by
+design. Something to keep chugging away until we do it right. It shouldn't
+take more work than 2 weeks.
+
+1. Code lives in ocs.
+1. URLs:
+   - Receiver of P1 submissions: phase1.gemini.edu/receive/.../...
+   - Viewer of P1 submissions: phase1.gemini.edu/fetch/.../...
+1. Each backend a different path like now.
+1. Configuration file for each backend (that file is read with every new
+   submission). Used for prepending semester to Ids, close deadline, etc.
+1. Id counter from previous file names (hashes discussed but discarded).
+1. Different backend for joints? If not possible use hard links.
+1. Generate PDFs.
+1. Stick together the XML and PDFs in tarballs.
+1. Send notifications *after* everything is successful. Consult configuration
+   file for emails.
+1. Multiple p1 versions:
+   a. When an XML is sent with an older PIT version, try to migrate to the latest
+      p1 model while parsing (as PIT does) and serialize it using the latest
+      version.
+   b. If there is a problem with the above, just change the Ids and add the
+      required tags by parsing directly the XML with an XML parsing library
+      (scala-xml + parsing combinators).
+1. Proposal viewer. XML stored in different directories with its own set of
+   HTTP basic authentication credentials. SSL should be possible.
```

Comment:
I'd rather not do it too, but I'm just reusing the HTTP auth that is already in place and we want something quick. We'd have to register the NGOs and all that, right?

Of course, in the long term we should use what we have in the Seqexec.